presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -410,6 +407,333 @@ Drop a schema::
+  may be low even if the whole system is heavy loaded.
+  See also :ref:`node-scheduler.network-topology<node-scheduler-properties>` if less
+  strict constrain is preferred - especially if some nodes are overloaded and other are not
+  fully utilized.
+
+
+``hive.max-initial-split-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``hive.max-split-size`` / ``2`` (``32 MB``)
+ * **Description:**
+
+  This property describes the maximum size of the first ``hive.max-initial-splits``
+  splits created for a query. the logic behind initial splits is described in
+  ``hive.max-initial-splits``. Lower values will increase concurrency for small queries.
+  This property represents the maximum size, as the real size may be lower when the amount
+  of data to read is less than ``hive.max-initial-split-size`` (e.g. at the end of a
+  block on a DataNode).
+
+
+``hive.max-initial-splits``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer``
+ * **Default value:** ``200``
+ * **Description:**
+
+  This property describes how many splits may be initially created for a single query
+  using ``hive.max-initial-split-size`` instead of ``hive.max-split-size``. A higher
+  value will force more splits to have a smaller size (``hive.max-initial-splits`` is
+  expected to be smaller than ``hive.max-split-size``), effectively increasing the
+  definition of what is considered a "small query". The purpose of the smaller split
+  size for the initial splits is to increase concurrency for smaller queries.
+
+
+``hive.max-outstanding-splits``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``1000``
+ * **Description:**
+
+  Limit on the nubmer of splits waiting to be served by a split source. After reaching
+  this limit, writers will stop writing new splits until some of hteme are used by workers.
+  Higher values will increase memory usage, but allow IO to be concentrated at one time,
+  which may be faster and increase resource utilization.
+
+
+``hive.max-partitions-per-writers``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``100``
+ * **Description:**
+
+  Maximum number of partitions per writer. A query will fail if it requires more partitions
+  per writer than allowed by this property. It can be helpful to have queries beyond the
+  expected maximum partitions to fail to help with error detection. Also it may allow to
+  preactivly avoid out of memory problem.
+
+
+``hive.max-split-iterator-threads``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``1000``
+ * **Description:**
+
+  This property describes how many threads may be used to iterate through splits when loading
+  them to the worker nodes. A higher value may increase parallelism, but increased concurrency
```

Comment:
I would drop: `A higher value may increase parallelism, but increased concurrency
  may cause too much time to be spent on context switching.`. It is obvious. 

Also here I would like to ask someone from FB about it, or dive into code.